src/main/java/com/ibm/stocator/fs/swift/SwiftAPIClient.java

```diff
@@ -483,17 +482,20 @@ public FSDataInputStream getObject(String hostName, Path path) throws IOExceptio
 
     LOG.debug("List container for {} container {}", obj, container);
     ArrayList<FileStatus> tmpResult = new ArrayList<FileStatus>();
-    PaginationMap paginationMap = cObj.getPaginationMap(obj, pageListSize);
-    FileStatus fs = null;
     StoredObject previousElement = null;
-    for (Integer page = 0; page < paginationMap.getNumberOfPages(); page++) {
-      Collection<StoredObject> res = cObj.list(paginationMap, page);
-      if (page == 0 && (res == null || res.isEmpty())) {
+    boolean moreData = true;
+    String marker = null;
+    FileStatus fs = null;
+    while (moreData) {
+      Collection<StoredObject> res = cObj.list(obj, marker, pageListSize);
+      moreData = (res.size() == pageListSize);
+      if (marker == null && (res == null || res.isEmpty() || res.size() == 0)) {
         FileStatus[] emptyRes = {};
         LOG.debug("List {} in container {} is empty", obj, container);
         return emptyRes;
       }
       for (StoredObject tmp : res) {
+        marker = tmp.getAsObject().getName();
```

Comment:
@effi-ofer what this does? It just update the new value in a loop? We need this to get the latest name to use it as a marker for the next read? - If so, may be we can do something like
this marker = res.get(res.size).getName() ?